src/components/Drawer/Drawer.js

```diff
@@ -0,0 +1,53 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import bem from 'bem';
+import { IoIosArrowUp } from 'react-icons/io';
+import styles from './Drawer.scss';
+
+const { block, elem } = bem({
+    name: 'Drawer',
+    classnames: styles,
+    propsToMods: ['isOpen'],
+});
+
+const Drawer = props => {
+    const { title, initialIsOpen, children, ...rest } = props;
+    const [isOpen, setIsOpen] = useState(initialIsOpen);
+
+    return (
+        <div {...block({ isOpen })} {...rest}>
+            <header {...elem('head', props)}>
+                <h4 {...elem('head-title', props)}>{title}</h4>
+                <section {...elem('head-buttons', props)}>
+                    <button
+                        {...elem('extend-button', { isOpen })}
+                        type="button"
+                        onClick={() => setIsOpen(!isOpen)}
```

Comment:
using of the arrow function can be avoided here

Reply:
For me not clear how we can avoid the arrow function. Could you please give an example?

Reply:
```
function handleClick() {
  setIsOpen(!isOpen);
}
. . .
onClick={handleClick}
```

Reply:
what is the profit use this approach and create an extra variable?

Reply:
In functional components, technically, it doesn't make much sense, cause this handler function will be recreated on every render. But for the sake of code style and better readability I'd still stick to the separate declaration of event handlers. Furthermore, with `useCallback` handler can be created only once.

```
function Foo() {
  const handleClick = useCallback(
    () => console.log('Click happened'),
    [],
  );
  return <Button onClick={handleClick}>Click Me</Button>;
}
```

Reply:
@mukiienko just to be consistent and transparent all over the code, so you would always expect a handler for an event inside the component. Also it's easier to extend the functionality inside the handler function this way.
It doesn't make any performance profit or whatsoever, but just unifies all of the handlers to follow the same pattern

Reply:
@antipin I think `useCallback` would be an overkill in a case like this but only makes it worse.
[Here](https://kentcdodds.com/blog/usememo-and-usecallback/) is an article which explains why